### PR TITLE
Fix handling of proc macro deps with multiple kinds

### PR DESCRIPTION
**Pull request by ulan**
A proc macro dependency may have multiple kinds such as build
and normal. Due to a bug in filtering predicate, such dependencies
are considered only as build dependency.

This PR fixes the bug and adds a regression test.

Issue: https://github.com/bazelbuild/rules_rust/issues/2210

### CODE REVIEW

File: crate_universe/src/metadata/dependency.rs [hunk 1]
```diff
@@ -604,6 +604,23 @@ mod test {
         );
     }
 
+    #[test]
+    fn multi_kind_proc_macro_dep() {
```
Comment (UebelAndre) [inline crate_universe/src/metadata/dependency.rs:608]: Can you also assert that the `build` dependency is correctly identified for build script targets?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thanks! One request 😄

File: crate_universe/src/metadata/dependency.rs (hunk 1 above)
Comment (ulan) [reply crate_universe/src/metadata/dependency.rs:608]: Done, thanks!

**Review by UebelAndre — APPROVED** [review]
Thank you!

**Comment by ulan** [comment]
@UebelAndre thanks for merging! Do you know approximately when the next version that includes this fix will be released?

**Pull request merged**